daffodil-cli/src/it/scala/org/apache/daffodil/CLI/Util.scala

```diff
@@ -200,4 +203,21 @@ object Util {
     }
     inputFile
   }
+
+  def expectExitCode(expectedExitCode: ExitCode.Value, shell: Expect): Unit = {
+    val expectedCode = expectedExitCode.id
+    val exitCodeCmd = if (Util.isWindows) "echo %errorlevel%" else "echo $?"
+    shell.sendLine(exitCodeCmd)
+
+    val sExitCode = shell.expect(contains("\n")).getBefore()
+    val actualInt = Integer.parseInt(sExitCode.trim())
```

Comment:
It looks like some tests are failing because sExitCode isn't an integer, it's like an error message or the empty string or something. My guess is this is because Daffodil just hasn't exited by the time we call this function, and so executing `echo ...` to display the error code won't run until after that finishes. But we're going to immediately start looking for a newline, so if Daffodil is still running and outputs a newline (which is very common), we'll think the exit code is just whatever Daffodil happened to output.

I wonder if maybe we need to do something like this:
```scala
val keyWord = "EXITCODE:"
val exitCodeCmd = "echo " + keyWord + (if (Util.isWindows) "%errorlevel%" else "$?")
shell.sendLine(exitCodeCmd)
shell.expect(contains(keyWord))
val sExitCode = shell.expect("\n").getBefore().trim()
val actualInt = Integer.parseInt(sExitCode)
...
```
So when we send the command to echo the exit code, we prepend a unique string. And then we then wait until we actually see this string. That way if Daffodil hasn't finished yet then we we'll wait until it is finished and the echo is actually run. Only once we see that do we then look for the newline and parse the int. This isn't tested, but something along these lines feels like the right approach.